elements/lisk-p2p/src/peer/base.ts

```diff
@@ -358,17 +359,22 @@ export class Peer extends EventEmitter {
 		);
 	}
 
-	public async fetchPeers(): Promise<ReadonlyArray<P2PPeerInfo>> {
+	public async fetchPeers(): Promise<ReadonlyArray<P2PEnhancedPeerInfo>> {
 		try {
 			const response: P2PResponsePacket = await this.request({
 				procedure: REMOTE_EVENT_RPC_GET_PEERS_LIST,
 			});
 
-			return validatePeerInfoList(
+			const validatedPeers = validatePeerInfoList(
 				response.data,
 				this._peerConfig.maxPeerDiscoveryResponseLength,
 				this._peerConfig.maxPeerInfoSize,
 			);
+
+			return validatedPeers.map(peerInfo => ({
```

Comment:
This looks like a sanitation of `P2PEnhancedPeerInfo`, isn't it? Shall we use `sanitizeEnhancedPeerInfo()` util?

Reply:
In this case it is adding `sourceAddress`. `sanitizeEnhancedPeerInfo` is to remove `sourceAddress` and other internal fields to peerBook.